### DISCUSSION TITLE
How add crypto.com wallet extension

### DISCUSSION THREAD

**Discussion by BahianDev** [rainbow-me/rainbowkit#1313] [Q&A]
`import` "@rainbow-me/rainbowkit/styles.css";
import type { AppProps } from "next/app";
import { ThemeProvider } from "styled-components";
import { theme } from "../styles/theme";
import GlobalStyle from "../styles/global";
import {
  getDefaultWallets,
  RainbowKitProvider,
  connectorsForWallets,
} from "@rainbow-me/rainbowkit";
import {
  injectedWallet,
  rainbowWallet,
  metaMaskWallet,
  coinbaseWallet,
  walletConnectWallet,
  phantomWallet,
} from "@rainbow-me/rainbowkit/wallets";
import { configureChains, createConfig, WagmiConfig } from "wagmi";
import { polygon, polygonMumbai } from "wagmi/chains";
import { jsonRpcProvider } from "wagmi/providers/jsonRpc";
import { publicProvider } from "wagmi/providers/public";
import { useEffect, useState } from "react";
import { Toaster } from "react-hot-toast";
import RoulleteProvider from "../contexts/roulette";
import ModalProvider from "../contexts/modal";

const { chains, publicClient } = configureChains([polygon], [publicProvider()]);

const connectors = connectorsForWallets([
  {
    groupName: "Phantom",
    wallets: [phantomWallet({ chains })],
  },
  {
    groupName: "Recommended",
    wallets: [
      injectedWallet({ chains }),
      rainbowWallet({ chains }),
      metaMaskWallet({ chains }),
    ],
  },
  {
    groupName: "Others",
    wallets: [
      coinbaseWallet({ chains, appName: "My RainbowKit App" }),
      walletConnectWallet({ chains }),
    ],
  },
]);

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient,
});

export default function App({ Component, pageProps }: AppProps) {
  const [ready, setReady] = useState(false);

  useEffect(() => {
    setReady(true);
  }, []);

  return (
    <>
      {ready ? (
        <ThemeProvider theme={theme}>
          <WagmiConfig config={wagmiConfig}>
            <RainbowKitProvider chains={chains}>
              <RoulleteProvider>
                <ModalProvider>
                  <Toaster
                    containerStyle={{
                      top: 95,
                    }}
                    toastOptions={{
                      className: "",
                      style: {
                        background: "#FFF",
                        color: "#000",
                        padding: "12px",
                        borderRadius: 0,
                      },
                    }}
                  />
                  <GlobalStyle />
                  <Component {...pageProps} />
                </ModalProvider>
              </RoulleteProvider>
            </RainbowKitProvider>
          </WagmiConfig>
        </ThemeProvider>
      ) : null}
    </>
  );
}
`

**Comment by DanielSinclair** [comment]
Even if a wallet is not directly supported by RainbowKit, you can create a [Custom Wallet](https://www.rainbowkit.com/docs/custom-wallets) connector and supply that to your [Custom Wallet List](https://www.rainbowkit.com/docs/custom-wallet-list). You can reference our wallet connector implementations [here](https://github.com/rainbow-me/rainbowkit/tree/main/packages/rainbowkit/src/wallets/walletConnectors) for more examples. RainbowKit's API was designed to be open for customization like this.
